src/mach/mod.rs

```diff
@@ -114,6 +115,37 @@ impl<'a> MachO<'a> {
             symbols::SymbolIterator::default()
         }
     }
+    pub fn compact_unwind_info(&self) -> error::Result<compact_unwind_info::CompactUnwindInfoIter> {
+        let get_unwind_info_section = || {
+            let section_name = "unwind_info";
+            for segment in &self.segments {
+                for section in segment {
+                    if let Ok((header, data)) = section {
+                        if let Ok(sec) = header.name() {
+                            if sec.len() >= 2 && &sec[2..] == section_name {
+                                // In some cases, dsymutil leaves sections headers but removes their
+                                // data from the file. While the addr and size parameters are still
+                                // set, `header.offset` is 0 in that case. We skip them just like the
+                                // section was missing to avoid loading invalid data.
+                                if header.offset == 0 {
+                                    return None;
+                                }
+
+                                return Some((header, data));
+                            }
+                        }
+                    }
+                }
+            }
+            None
+        };
+
+        if let Some((_header, data)) = get_unwind_info_section() {
+            compact_unwind_info::CompactUnwindInfoIter::new(data, self.little_endian, self.is_64)
+        } else {
+            Err(error::Error::Malformed(String::from("No unwind_info section found")))
```

Comment:
Maybe this should just return option instead of malformed? Or are unwind_info sections required to be present in a macho binary?

Reply:
Yes that's a good idea. At the time it felt strange to have `Result<Option<...>>` but now it's totally reasonable to me.